### PR TITLE
Develop beneficiary governance

### PR DESCRIPTION
**Pull request by anthonymartin**

### CODE REVIEW

File: packages/app/components/Proposals/Form/Preview.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                setCurrentStep(currentStep - 1);
+              }}
+            >
+              Back/Edit
+            </button>
+            <button
+              type="button"
+              className="ml-3 inline-flex items-center px-4 py-2 border border-transparent rounded-md shadow-sm text-sm font-medium text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
+              onClick={() => {
+                // FIXME: Promise ignored
```
Comment (anthonymartin) [inline packages/app/components/Proposals/Form/Preview.tsx:172]: @mefuru could you write a ticket for this todo?

File: netlify.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     mode = 'prefix'
 
 [context.deploy-preview]
-  command = "cd packages/contracts; yarn compile; cd ../../packages/app; yarn build;"
+  command = "cd packages/utils; yarn build; cd ../../packages/contracts; yarn compile; cd ../../packages/app; yarn build;"
```
[the remark this answers is not in the document]
Comment (mefuru) [reply netlify.toml]: don't we need to run the TypeScript compiler on utils package?

File: packages/app/components/Grants/GrantFunded.tsx [hunk 3]
```diff
@@ -1,41 +1,53 @@
...
+    ).map((a) => a.toLowerCase());
+    if (awarded.includes(beneficiary.beneficiaryAddress)) {
       setAwarded(true);
     }
-  }
+  };
 
   useEffect(() => {
     if (contracts?.grant) {
+      // FIXME: Promise being ignored
```
Comment (anthonymartin) [inline packages/app/components/Grants/GrantFunded.tsx:32]: @mefuru do you know what this is about?

File: packages/app/components/Grants/GrantRound.tsx [hunk 4]
```diff
@@ -50,19 +52,28 @@ export default function GrantRound({
...
+  };
+
   useEffect(() => {
     if (votes && election) {
-      setBeneficiaries(
-        election.registeredBeneficiaries.map((address) =>
-          getBeneficiary(address, votes),
-        ),
-      );
+      // FIXME: Promise being ignored
```
Comment (anthonymartin) [inline packages/app/components/Grants/GrantRound.tsx:75]: @mefuru  and this too

File: packages/app/components/Grants/GrantFunded.tsx (hunk 3 above)
Comment (mefuru) [reply packages/app/components/Grants/GrantFunded.tsx:32]: No, I don't. They were added in the PR where components were changed to use React.FC type

File: packages/app/components/Grants/GrantRound.tsx (hunk 4 above)
Comment (mefuru) [reply packages/app/components/Grants/GrantRound.tsx:75]: same as above

File: packages/app/components/Proposals/Form/DisplayFiles.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  localState: string | string[];
+}
+
+export const DisplayImage: React.FC<DisplayFilesProps> = ({ localState }) => {
+  return (
+    <div key={localState as string}>
+      {localState?.length && (
+        <img
+          className="mx-auto w-1/2"
+          src={'https://gateway.pinata.cloud/ipfs/' + localState}
```
Comment (anthonymartin) [inline packages/app/components/Proposals/Form/DisplayFiles.tsx:11]: replace hardcoded ipfs url with ENV var

**Review by mefuru — APPROVED** [review]

File: packages/app/components/Proposals/ProposalGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                name="status"
+                className="appearance-none block w-full bg-none bg-white border border-gray-300 rounded-md pl-3 pr-10 py-2 text-base text-gray-900 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
+                defaultValue={'All'}
+                value={ProposalStatus[statusFilter]}
+                onChange={(e) => {
+                  const status = e.target.value;
+                  setStatusFilter(ProposalStatus[status]);
+                }}
+              >
+                {new Array(6).fill(undefined).map((x, status) => {
```
Comment (anthonymartin) [inline packages/app/components/Proposals/ProposalGrid.tsx:124]: replace this with `Object.keys(ProposalStatus).map` for dynamic display

**Review by mefuru — COMMENTED** [review]
Added #303 and #304

**Pull request merged**